packages/rich-text/src/get-active-format.native.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * External dependencies
+ */
+
+import { find } from 'lodash';
+
+/**
+ * Gets the format object by type at the start of the selection. This can be
+ * used to get e.g. the URL of a link format at the current selection, but also
+ * to check if a format is active at the selection. Returns undefined if there
+ * is no format at the selection.
+ *
+ * @param {Object} value      Value to inspect.
+ * @param {string} formatType Format type to look for.
+ *
+ * @return {?Object} Active format object of the specified type, or undefined.
+ */
+export function getActiveFormat( { formats, formatPlaceholder, start, end }, formatType ) {
+	if ( start === undefined ) {
+		return;
+	}
+
+	// if selection is not empty, get the first character format
+	if ( start !== end ) {
+		return find( formats[ start ], { type: formatType } );
+	}
+
+	// if user picked (or unpicked) formats but didn't write anything in those formats yet return this format
+	if ( formatPlaceholder && formatPlaceholder.index === start ) {
+		return find( formatPlaceholder.formats, { type: formatType } );
+	}
+
+	// otherwise get the previous character format
+	const previousLetterFormat = find( formats[ start - 1 ], { type: formatType } );
```

Comment:
What do you think about adding a case here for the start of text? I mean, when we're at the index 0, we should just pick up the formats from the first char available. Something like this:

```suggestion
	// if we're at the start of text, use the first char to pick up the formats
	const startPos = start == 0 ? 0 : start -1;

	// otherwise get the previous character format
	const previousLetterFormat = find( formats[ startPos ], { type: formatType } );
```

This will fix the problem of not highlighting the "B" button when placing the caret at the beginning of text when that text has a bold word to start.

Reply:
I will go ahead and introduce the change so to match Aztec-standalone and Gutenberg-web's behavior.

Reply:
Done with b0fd2c0.